### PR TITLE
docs(root): add required status to props description in props table 

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes

Add required status to props table on component code pages. Add attribute to prop name column. Sort props table so required props appear first

## Related issue

#149,  #233

## Checklist

- [x] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Review by GCHQ-Developer-112 — CHANGES_REQUESTED** [review]
Hey, there's currently a double padding being applied on the prop descriptions and so they no longer match the designs

Screenshot 2023-01-18 at 08 58 32

**Comment by GCHQ-Developer-530** [comment]
> Hey, there's currently a double padding being applied on the prop descriptions and so they no longer match the designs
> 
> **Website** Screenshot 2023-01-18 at 08 58 32
> 
> **Designs** Screenshot 2023-01-18 at 09 08 52



> Hey, there's currently a double padding being applied on the prop descriptions and so they no longer match the designs
> 
> **Website** Screenshot 2023-01-18 at 08 58 32
> 
> **Designs** Screenshot 2023-01-18 at 09 08 52

This is an issue on the old build of this branch, unfortunately there's some weird issue with the build so the fixed padding hasn't come through yet. It now looks like this:
Screenshot 2023-01-18 at 08 57 25

**Comment by GCHQ-Developer-530** [comment]
The build should now be working :)

File: src/components/ComponentDetails/PropTable/PropDescription/index.tsx [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
-}) => (
-  <div>
+const PropDescription: React.FC<{
+  description: string;
+  type: string;
+  required: boolean;
+}> = ({ description, type, required }) => (
+  <div className="prop-description">
     <ic-typography>{description}</ic-typography>
+    {required === true && (
```
Comment (jd239) [inline src/components/ComponentDetails/PropTable/PropDescription/index.tsx:12]: You shouldnt need the `=== true` as the type is a boolean 
This should do the job already `{required && (`

File: src/components/ComponentDetails/PropTable/index.tsx [hunk 2]
```diff
@@ -38,7 +39,11 @@ const PropTable: React.FC<PropTableProps> = ({ propData }) => {
       propData.map((prop) => ({
```
Comment (jd239) [inline src/components/ComponentDetails/PropTable/index.tsx:39]: Let's destructure props so passing the props is a bit cleaner
`propData.map({ docs, type, required }) => ({`
`<PropDescription description={docs} type={type} required={required} />`

File: src/components/ComponentDetails/PropTable/PropDescription/index.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.prop-description .ic-typography-subtitle-small::before {
+  color: var(--ic-status-error);
+  content: "*";
+}
```
Comment (jd239) [inline src/components/ComponentDetails/PropTable/PropDescription/index.css:4]: I'm assuming this has already been discussed but shouldn't the word required also be red to draw attention to it being required?

**Review by jd239 — CHANGES_REQUESTED** [review]
Looks good. Small issues. Nothing crazy.

File: src/components/ComponentDetails/PropTable/PropDescription/index.css (hunk 3 above)
Comment (GCHQ-Developer-530) [reply src/components/ComponentDetails/PropTable/PropDescription/index.css:4]: In the figma file it only showed the asterisk as red, so I followed that.
Screenshot 2023-01-20 at 11 32 06

**Review by DBD324 — CHANGES_REQUESTED** [review]
The required styling looks good, but can you double check the vertical spacing against the updated designs in the Figma 2.1 component library. I think this developed version is a bit tight in comparison. Thanks

**Review by jd239 — DISMISSED** [review]

File: src/components/ComponentDetails/PropTable/PropDescription/index.tsx [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
-    <IcTypography>{description}</IcTypography>
+const PropDescription: React.FC<{
+  description: string;
+  type: string;
+  required: boolean;
+}> = ({ description, type, required }) => (
+  <div className="prop-description">
+    <ic-typography>{description}</ic-typography>
+    {required && (
+      <ic-typography variant="subtitle-small">required</ic-typography>
```
Comment (GCHQ-Developer-112) [inline src/components/ComponentDetails/PropTable/PropDescription/index.tsx:13]:
```suggestion
      <ic-typography variant="subtitle-small">Required</ic-typography>
```
Small change, but the designs show "Required" with a capital R
Screenshot 2023-01-24 at 18 46 22

**Comment by GCHQ-Developer-112** [comment]
The horizontal spacing between the different parts of the props card doesn't look quite right in comparison to the designs

Designs
Group 1

Developed components
Group 2

**Comment by GCHQ-Developer-112** [comment]
> The horizontal spacing between the different parts of the props card doesn't look quite right in comparison to the designs
> 
> Designs Group 1
> 
> Developed components Group 2

I've made these changes to the PR

**Comment by DBD324** [comment]
Looks like there is still a padding issue around the 'Required' label on larger screens. The Required label has 4px top and bottom margins applied causing the gap between the elements to be 12px instead of 8. The spacing token used in the gap is correct; xs spacing (8px). 

image

**Comment by DBD324** [comment]
Additionally, the 'type' label background is only 28px high when it should be 32px high. 
image

**Review by DBD324 — CHANGES_REQUESTED** [review]
Added comments for further spacing changes. On all breakpoints the description elements are not spaced according to the designs.

**Comment by GCHQ-Developer-112** [comment]
> Additionally, the 'type' label background is only 28px high when it should be 32px high. image

Hey, the padding around the type text matches the designs. I believe the reason that it's not 32px is because we are not using the code typography variant (which currently doesn't exist).  Would you still like the type label to be 32px considering this?

**Comment by DBD324** [comment]
On XS breakpoint, the spacing between sections in the card view uses 'md' spacing token (16px) instead of 'lg' (24px). 
Also, the same issues with the description, as mentioned in comments above, exist on this breakpoint too.

**Comment by GCHQ-Developer-847** [comment]
A bit out of scope for this ticket but it looks like the gap between the columns is 48px because of the 24px on both sides - whereas in the designs it's just 24px in total. Might be worth fixing in this PR?
image
image

**Review by DBD324 — APPROVED** [review]
Looks great. Thanks for making the last few changes.

**Comment by GCHQ-Developer-112** [comment]
> A bit out of scope for this ticket but it looks like the gap between the columns is 48px because of the 24px on both sides - whereas in the designs it's just 24px in total. Might be worth fixing in this PR?
> 
> image image

Done 😄

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Review by GCHQ-Developer-112 — APPROVED** [review]

**Pull request merged**